sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -921,11 +850,6 @@ abstract class CastSuiteBase extends SparkFunSuite with ExpressionEvalHelper {
     Seq("nan", "nAn", " nan ").foreach { value =>
       checkEvaluation(cast(value, DoubleType), Double.NaN)
     }
```

Comment:
`cast('nan' as float)` is working fine.
Although `CAST(CAST(nan AS DECIMAL(10,0)) AS DOUBLE) ` is returning `NaN` in pgsql.